app/controllers/submissions_controller.rb

```diff
@@ -11,7 +11,19 @@ def update
     params[:material_submission][:status] = step.to_s
     params[:material_submission][:status] = 'active' if last_step?
 
-    if material_submission.update_attributes(material_submission_params) && last_step?
+    updating_params = material_submission_params
+    case step
+    when :dispatch
+      if material_submission_params[:contact].empty?
+        flash[:error] = "You need to provide a contact"
+        redirect_to material_submission_build_path material_submission_id: material_submission.id
+        return
+      else
+        set_contact
```

Comment:
I think the setting of a contact may belong in the MaterialSubmission model, which would mean you wouldn't have to do any validation in the this controller method. Will discuss in person.